### PR TITLE
Sohaib Frontend searchbar,Pagination,Create Librarian

### PR DESCRIPTION
**Pull request by SohaibkhanFolio3**
In this PR  , Librarian can only Create Librarian , Pagination is added on Books , Authors, Genres , Search bar to filter books is also implemented
Tickets:
https://app.asana.com/0/1204411152293714/1204647902309401/f
https://app.asana.com/0/1204411152293714/1204647902309389/f
https://app.asana.com/0/1204411152293714/1204647902309391/f

### CODE REVIEW

File: frontend/src/axios.tsx [hunk 1]
```diff
@@ -439,6 +440,22 @@ export class APIClient {
...
+  public LibrarianSignup(data: SignupData): Promise<Tokens> {
+    return APIClient.axiosInstance.post("/auth/librarian/register", data)
+      .then((res) => {
+        let tokens: Tokens = {
+          access_token: res.data.data.access_token,
+          refresh_token: res.data.data.refresh_token,
+        };
+        localStorage.setItem("access_token", tokens.access_token);
+        localStorage.setItem("refresh_token", tokens.refresh_token);
+        APIClient.instance.SetAuthorizationHeaders(
```
Comment (mustansirMuz) [inline frontend/src/axios.tsx:452]: We do not need to change the authentication header here. This basically means that we will be using the token of the newly created librarian now onwards, but that is not the case

**Review by mustansirMuz — CHANGES_REQUESTED** [review]
See requested change

File: frontend/src/axios.tsx (hunk 1 above)
Comment (mtalhatahir2001) [reply frontend/src/axios.tsx:452]: Humm, Similarly I also don't think that we need to replace token in local storage for the same reason @mustansirMuz  mentioned.

**Review by mtalhatahir2001 — CHANGES_REQUESTED** [review]
It needs some minor fixes.

File: frontend/src/axios.tsx (hunk 1 above)
Comment (SohaibkhanFolio3) [reply frontend/src/axios.tsx]: DONE

**Review by mustansirMuz — APPROVED** [review]

**Review by mtalhatahir2001 — APPROVED** [review]
LGTM

**Pull request merged**